src/Sinch/Core/Http.cs

```diff
@@ -80,23 +86,87 @@ internal class Http : IHttp
                 $"sinch-sdk/{sdkVersion} (csharp/{RuntimeInformation.FrameworkDescription};;)";
         }
 
+        public Task<TResponse> SendMultipart<TRequest, TResponse>(Uri uri, TRequest request, Stream stream,
+            string fileName, CancellationToken cancellationToken = default)
+        {
+            var content = new MultipartFormDataContent();
+            var props = request!.GetType().GetProperties(BindingFlags.Instance | BindingFlags.Public |
+                                                         BindingFlags.DeclaredOnly)
+                .Where(DoesntHaveJsonIgnoreAttribute).Where(HasNonNullValue);
+            foreach (var prop in props)
+            {
+                var value = prop.GetValue(request);
+                if (value == null)
+                {
+                    continue;
+                }
+
+                var type = value.GetType();
+                if (type == typeof(List<string>))
+                {
+                    var asString = string.Join(',', (value as List<string>)!);
+                    content.Add(new StringContent(asString), prop.Name);
+                }
+                else if (type == typeof(Dictionary<string, string>))
+                {
+                    foreach (var (key, val) in (value as Dictionary<string, string>)!)
```

Comment:
I would have expected to see a recursion for such type. How do you handle nested structures?